### PR TITLE
Add GPU tests, plus tests for bathymetry module

### PR DESCRIPTION
**Pull request by simone-silvestri**
follows the detailes outlined in #141

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -54,5 +54,13 @@ steps:
     commands:
       - "julia --project -e 'using Pkg; Pkg.test()'"
 
+  - label: "Run bathyemtry tests"
+    key: "tests_bathymetry"
+    env:
+      CUDA_VISIBLE_DEVICES: "-1"
```
Comment (glwagner) [inline .buildkite/pipeline.yml:60]: Probably we should test on the GPU right?

**Review by glwagner — APPROVED** [review]

File: test/test_bathymetry.jl [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                                     latitude = (0, 50),
+                                     z = (-6000, 0))
+
+        # Test that remove_minor_basins!(Z, Inf) does nothing
+        bottom_height = regrid_bathymetry(grid)        
+        control_bottom_height = deepcopy(bottom_height)
+
+        remove_minor_basins!(bottom_height, Inf)
+
+        @test interior(bottom_height) .== interior(control_bottom_height)
```
Comment (glwagner) [inline test/test_bathymetry.jl:20]:
```suggestion
        bottom_height = regrid_bathymetry(grid)        
        control_bottom_height = deepcopy(bottom_height)
        remove_minor_basins!(bottom_height, Inf)
        @test interior(bottom_height) .== interior(control_bottom_height)
```

File: test/test_bathymetry.jl [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        # A two basins bathymetry
+        bottom(x, y) = - 1000 * Int((x < 10) | (x > 45)) 
+        
+        set!(bottom_height, bottom)
+        set!(control_bottom_height, bottom)
+
+        # This should have not changed anything
+        remove_minor_basins!(bottom_height, 2)
+
+        @test interior(bottom_height) .== interior(control_bottom_height)
```
Comment (glwagner) [inline test/test_bathymetry.jl:35]:
```suggestion
        remove_minor_basins!(bottom_height, 2)
        @test interior(bottom_height) .== interior(control_bottom_height)
```

File: test/test_bathymetry.jl [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                                     latitude = (-10, 50),
+                                     z = (-6000, 0))
+
+        control_bottom_height = regrid_bathymetry(grid)
+        interpolated_bottom_height = regrid_bathymetry(grid; interpolation_passes = 100)
+
+        # Testing that multiple passes do not change the solution when refining the grid
+        @test interior(control_bottom_height) .== interior(interpolated_bottom_height)
+    end
+end
```
Comment (glwagner) [inline test/test_bathymetry.jl:54]:
```suggestion

end 
```

**Comment by simone-silvestri** [comment]
I have added the gpu tests using an environmental variable. I remember we decided to go in that direction, but I forgot which issue we were discussing about GPU tests. I can change it following a different strategy because I cannot find out which was the issue.

File: .buildkite/pipeline.yml [hunk 5]
```diff
@@ -22,45 +22,114 @@ steps:
...
       TEST_GROUP: "fluxes"
     commands:
       - "julia --project -e 'using Pkg; Pkg.test()'"
+    agents:
+      slurm_mem: 32G
+      slurm_cpus_per_task: 8
+      slurm_ntasks: 1
 
-  - label: "Run bathyemtry tests"
+  - label: "CPU bathymetry tests"
```
Comment (glwagner) [inline .buildkite/pipeline.yml:65]: Is it best to use a separate job for CPU vs GPU tests, or just do both CPU and GPU tests in one job?

File: .buildkite/pipeline.yml [hunk 6]
```diff
@@ -77,57 +77,57 @@ steps:
     key: "gpu_jra55_tests"
     env:
       TEST_GROUP: "jra55"
+      GPU_TEST: "true"
```
Comment (glwagner) [inline .buildkite/pipeline.yml:80]: A bit more flexible would be something like

```suggestion
      OCEANANIGANS_ARCHITECTURE: "GPU"
```

I suppose "false" is meant to signify CPU. But explicitly denoting the architecture might allow additional possibilities.

File: test/test_bathymetry.jl [hunk 7]
```diff
@@ -18,7 +18,7 @@ using ClimaOcean.Bathymetry: remove_minor_basins!
 
         # A fictitiously large number which should presumably keep all the basins
         remove_minor_basins!(bottom_height, 10000000) 
-        @test interior(bottom_height) .== interior(control_bottom_height)
+        @test all(interior(bottom_height) .== interior(control_bottom_height))
```
Comment (glwagner) [inline test/test_bathymetry.jl:21]:
```suggestion
        @test parent(bottom_height) == parent(control_bottom_height)
```

does this work?

**Comment by glwagner** [comment]
CUDA initialization doesn't seem to be working, since https://buildkite.com/clima/climaocean-ci/builds/1215#01916d39-cf52-4943-a59e-d63e89f92b49/206-472

**Pull request merged**